packages/teams-js/src/public/liveShareHost.ts

```diff
@@ -132,6 +132,19 @@ export interface IFluidTenantInfo {
   storageEndpoint: string;
 }
 
+/**
+ * @hidden
+ * Returned from `LiveShareHost.getClientInfo()` to specify the client info for a
+ * particular client in a Live Share session.
+ *
+ * @beta
+ */
+export interface IClientInfo {
+  userId: string;
```

Comment:
Guid

Reply:
(But a stronger type would be much better and easier than that so I would recommend that one)

Reply:
in utils.ts i see some uuid utils, but they are not using a uuid type. Everything looks to just be a string. There isn't a validate method available until later versions of the uuid lib, but i'm not sure validating matters here much, as live-share-sdk doesn't care about the format, and the type would still be exposed to the user as a string and not a uuid type.

Reply:
This library has in a lot of places, made the mistake of not being strict about types and uuid is one of those mistakes. We've run into several other places in the library where this exact issue has bitten us (things like assuming Ids will be globally unique and then finding out that they definitely aren't). If this is actually a UUID, we should make sure to convert it to a UUID type and return that to app developers to that they can make good decisions about how to treat them. It will also make it easier for other hosts to integrate with this functionality in the future and pass consistent output out to app developers. Please feel free to add a uuid type such as https://www.npmjs.com/package/@types/uuid

Reply:
@TrevorJoelHarris ss far as I can tell there is no uuid type that we can replace the string type with.
Here is a link to the [uuid-test.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/uuid/v7/uuid-tests.ts) file of the @types/uuid package, which shows usage of it.

I think the purpose of that package is to make calling the uuid functions type safe, but we would still be working with string output.

Reply:
It seems like it should be easy enough (famous last words) to just create a small custom `Uuid` type that leverages that library to maintain type safety. We (teamsjs) should probably create a robust type that can be used throughout, but just to bootstrap it, I'm guessing you could create a small one that, at least, validates in its constructor that it's been given a valid uuid and works just for your scenarios?

E.g., like the answers (or even a subset of them) on this StackOverflow post?
https://stackoverflow.com/questions/37144672/guid-uuid-type-in-typescript

Reply:
@AE-MS, @TrevorJoelHarris So in this case the userId is a Microsoft GUID not a UUID. Turns out there are some subtle differences 😞 

I don't think it makes a bunch of sense to have a GUID class be part of teams-js. Ideally we would have a separate Microsoft GUID package compatible with the [c# implementation](https://github.com/microsoft/referencesource/blob/master/mscorlib/system/guid.cs), that we would use in all typescript projects, but that doesn't seem to exist right now.

In my case, Live-Share-SDK doesn't take a dependency on teams-js. The LiveShareHost in teams-js matches the interface declared in the Live-Share-SDK through duck typing. We would have to duplicate the same GUID class there in order to use it. 

This would also go for GUIDs coming from the teams client, which also doesn't take a dependency on teams-js. All of the GUID's there are represented as strings.

in the meantime, instead of a GUID class, would you be okay with a type brand, which would allow the user to know the type, but also continue to be able to use it like a string?
```typescript
export type Guid = string & { _guidBrand: undefined };
```

I've pushed a commit just to show what that would look like.

Reply:
if we did want to go with the GUID route i could put together something together like this.
```typescript
// this would be duplicated in live-share-sdk
export class Guid {
    private static validator = new RegExp(
        "^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$",
        "i"
    );

    private value: string;

    public constructor(value: string) {
        if (Guid.validator.test(value)) {
            this.value = value;
        } else {
            throw new Error(`invalid Guid: ${value}`);
        }
    }

    public toString(): string {
        return this.value;
    }
}
```


```typescript
  public getClientInfo(clientId: string): Promise<IClientInfo | undefined> {
    ensureSupported();
    return new Promise<IClientInfo | undefined>((resolve) => {
      resolve(
        sendAndHandleSdkError('interactive.getClientInfo', clientId).then(
          // represents what we would get from the teams client
          (info: { userId: string; roles: UserMeetingRole[]; displayName?: string }) => {
            return {
              userId: new Guid(info.userId),
              roles: info.roles,
              displayName: info.displayName,
            };
          },
        ),
      );
    });
  }
```

Reply:
@AE-MS, @TrevorJoelHarris It turns out we **_cannot guarantee it is always a GUID_**. While it will always be a GUID when it comes from teams, the LiveShareHost was designed to be generic, so if someone else implemented a host, userId could be of a different format entirely. 

We have to leave it as a string in this case, i'll be reverting my last commit.